web/src/helpers/nodes.ts

```diff
@@ -14,3 +14,42 @@ export function isDataset(node: MqNode): Undefinable<LineageDataset> {
   }
   return undefined
 }
+
+export function encodeNode(nodeType: JobOrDataset, namespace: string, name: string) {
+  return `${encodeURIComponent(nodeType.toLowerCase())}/${encodeURIComponent(
+    namespace
+  )}/${encodeURIComponent(name)}`
+}
+
+export function generateNodeId(type: JobOrDataset, namespace: string, jobName: string) {
+  return `${type.toLowerCase()}:${namespace}:${jobName}`
+}
+
+export function isLineageJob(
+  node: Undefinable<LineageJob | LineageDataset>
+): Undefinable<LineageJob> {
+  if (node && node.type === 'BATCH') {
```

Comment:
`STREAM` and `STREAM` are also job types.

Reply:
Yeah, we can adjust this in follow-up work.